enterprise/app/executors/executors.tsx

```diff
@@ -288,47 +292,76 @@ export default class ExecutorsComponent extends React.Component<Props, State> {
     this.fetchExecutors();
   }
 
+  onClickTab(tabId: TabId) {
+    router.navigateTo(`/executors/${tabId}`);
+  }
+
   // "bring your own runners" is enabled for the installation (i.e. BuildBuddy Cloud deployment).
   renderWithGroupOwnedExecutorsEnabled() {
-    if (this.props.user?.selectedGroup?.useGroupOwnedExecutors) {
-      if (this.state.nodes.length == 0) {
-        return (
-          <div className="empty-state">
-            <h1>No self-hosted executors connected.</h1>
-            <p>Self-hosted executors are enabled, but there are no executors connected.</p>
-            <p>Follow the instructions below to deploy your executors.</p>
-            <ExecutorSetup
-              user={this.props.user}
-              executorKeys={this.state.executorKeys}
-              executors={this.state.nodes}
-              schedulerUri={this.state.schedulerUri}
-            />
+    const linuxNodes = this.state.nodes.filter((node) => node.os.toLowerCase() === "linux");
```

Comment:
I don't love the idea of the front-end having to know about special case linux vs mac executor logic.

Also, whether or not this special case logic applies depends on whether or not this config option is set: https://github.com/buildbuddy-io/buildbuddy/blob/master/server/config/config.go#L255

Maybe GetExecutionNodes could return something like whether or not the returned executor is enabled?

Reply:
SG -- I added an "enabled" bit and surfaced it in a new proto `GetExecutionNodesResponse.Executor` -- I didn't add the bit to the `ExecutionNode` because currently that proto just contains static configuration that gets stored in Redis, and don't want ppl to accidentally rely on that enabled bit for anything other than the executors page.

For now the UI assumes that if at least one executor is disabled, then it's because self-hosted executors are not enabled in settings. If there are other reasons later on, we can maybe change this to a `status` enum or add a `disabled_reason` enum or something like that.

![image](https://user-images.githubusercontent.com/2414826/145083208-08ca2ca0-7e8e-44ef-99d3-79ba34e24ed1.png)